### PR TITLE
Correct epoll lock usage

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary

- vfs/epoll: Replace sem_t with mutex_t for locking usage 
- vfs/epoll: Protect epoll_ctl by mutex 

## Impact
epoll

### CODE REVIEW

File: fs/vfs/fs_epoll.c [hunk 1]
```diff
@@ -277,6 +278,7 @@ void epoll_close(int epfd)
 int epoll_ctl(int epfd, int op, int fd, struct epoll_event *ev)
```
Comment (pkarashchenko) [inline fs/vfs/fs_epoll.c:278]:
```suggestion
int epoll_ctl(int epfd, int op, int fd, FAR struct epoll_event *ev)
```

File: fs/vfs/fs_epoll.c [hunk 2]
```diff
@@ -352,19 +360,26 @@ int epoll_ctl(int epfd, int op, int fd, struct epoll_event *ev)
...
 
+  nxmutex_unlock(&eph->lock);
   poll_notify(&eph->poll, 1, POLLIN);
   return 0;
+
+err:
+  nxmutex_unlock(&eph->lock);
+err_without_lock:
+  set_errno(-ret);
+  return -1;
```
Comment (pkarashchenko) [inline fs/vfs/fs_epoll.c:382]: should we use `ERROR` and `OK`?

File: fs/vfs/fs_epoll.c [hunk 3]
```diff
@@ -201,13 +203,12 @@ static int epoll_do_create(int size, int flags)
   fd = file_allocate(&g_epoll_inode, flags, 0, eph, 0, true);
   if (fd < 0)
     {
-      nxsem_destroy(&eph->sem);
+      nxmutex_destroy(&eph->lock);
       kmm_free(eph);
       set_errno(-fd);
       return -1;
```
Comment (pkarashchenko) [inline fs/vfs/fs_epoll.c:209]: Should we use `ERROR` and `OK`?

File: fs/vfs/fs_epoll.c (hunk 2 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_epoll.c]: Done.

File: fs/vfs/fs_epoll.c (hunk 1 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_epoll.c]: Done.

File: fs/vfs/fs_epoll.c (hunk 3 above)
Comment (xiaoxiang781216) [reply fs/vfs/fs_epoll.c]: Done.

**Review by pkarashchenko — APPROVED** [review]

File: fs/vfs/fs_epoll.c [hunk 4]
```diff
@@ -180,10 +181,11 @@ static int epoll_do_create(int size, int flags)
   if (eph == NULL)
     {
       set_errno(ENOMEM);
-      return -1;
+      return ERROR;
     }
 
-  nxsem_init(&eph->sem, 0, 0);
```
Comment (pkarashchenko) [inline fs/vfs/fs_epoll.c:186]: The new behavior is a bit different from the original one. I do not think that should matter, but in "old" code the semaphore kept locked after init and only unlocked before return. Do we need to take lock and hold until `file_allocate` return `OK`?

Comment (xiaoxiang781216) [reply fs/vfs/fs_epoll.c:186]: It isn't a problem since the user can't do anything before we return fd from epoll_do_open.

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**